### ISSUE TITLE
[share-online] Download does not start

### ISSUE THREAD

**Issue by ironbiff** [pyload/pyload#3035]
Hi there, I am not sure if it is a bug or my mistake but I am unable to resolve this issue.

- Installed new Docker Pyload
https://hub.docker.com/r/writl/pyload/
- setup with share-online account

pyload creates 13KB files for each "started" download and set timer to 5 minutes although I have a premium account for share-online
"Waiting 5 minutes..."
after this it restarts the download

see full Log with debug
[log.txt](https://github.com/pyload/pyload/files/1522879/log.txt)

**Comment by GammaC0de** [comment]
I have tested it with docker and it worked fine with the URL in the log (free account, not premium)

How do yu rin the pyload docker? it should be run something like
```
docker run -d -v /home/user/pyload/downloads:/opt/pyload/Downloads -v /home/user/pyload/config:/opt/pyload/pyload-config -P writl/pyload:latest
```

Also check that you have correct permissions to the download and config directories

**Comment by ironbiff** [comment]
Thank you for your answer and testing.

Yes I run it like this. Just with different folders on the outside of the container. I used it for several month like this and than I decided to reload the container to update pyload. Never stop a running system... ;( 

Permissions should be fine because everything is running in root context and pyload is able to write in the config directory and in the download directory.

I will reset it completely inclusive the config and will respond soon

**Comment by ironbiff** [comment]
OK, tried it with default settings and as you said, everything works like a charm.
- deleted download
- Added the share-online account
- added the download again
= Error arise again 


97 | 02.12.2017 13:28:58 | INFO | HOSTER ShareonlineBiz[2]: Grabbing link info...
-- | -- | -- | --
98 | 02.12.2017 13:28:58 | INFO | HOSTER ShareonlineBiz[2]: Link name: TVNowDownloader_Portable_v1.1.0.1.rar
99 | 02.12.2017 13:28:58 | INFO | HOSTER ShareonlineBiz[2]: Link size: 18.20 MiB (19088982 bytes)
100 | 02.12.2017 13:28:58 | INFO | HOSTER ShareonlineBiz[2]: Link status: online
101 | 02.12.2017 13:28:58 | INFO | HOSTER ShareonlineBiz[2]: Processing url: http://www.share-online.biz/dl/U4W59IZO1GT
102 | 02.12.2017 13:28:58 | INFO | HOSTER ShareonlineBiz[2]: Looking for direct download link...
103 | 02.12.2017 13:28:58 | INFO | HOSTER ShareonlineBiz[2]: Direct download link not found
104 | 02.12.2017 13:28:59 | INFO | ACCOUNT ShareonlineBiz: Grabbing account info for user `238*******`...
105 | 02.12.2017 13:28:59 | INFO | HOSTER ShareonlineBiz[2]: Filesize: 18641 KiB \| Traffic left for user `238*******`: 104857600 KiB
106 | 02.12.2017 13:28:59 | INFO | HOSTER ShareonlineBiz[2]: Processing as premium download...
107 | 02.12.2017 13:28:59 | INFO | ACCOUNT ShareonlineBiz: Grabbing account info for user `238*******`...
108 | 02.12.2017 13:28:59 | INFO | HOSTER ShareonlineBiz[2]: Downloading file...
109 | 02.12.2017 13:29:00 | INFO | HOSTER ShareonlineBiz[2]: File saved
110 | 02.12.2017 13:29:00 | INFO | HOSTER ShareonlineBiz[2]: Checking download...
111 | 02.12.2017 13:29:00 | INFO | HOSTER ShareonlineBiz[2]: File is OK
112 | 02.12.2017 13:29:00 | INFO | HOSTER ShareonlineBiz[2]: Waiting 5 minutes...
113 | 02.12.2017 13:34:02 | INFO | Download restarted: TVNowDownloader_Portable_v1.1.0.1.rar \| Download failed
114 | 02.12.2017 13:34:02 | INFO | Download starts: TVNowDownloader_Portable_v1.1.0.1.rar
115 | 02.12.2017 13:34:02 | WARNING | HOSTER ShareonlineBiz[2]: Plugin may be unstable
116 | 02.12.2017 13:34:02 | INFO | ACCOUNT ShareonlineBiz: Grabbing account info for user `238*******`...
117 | 02.12.2017 13:34:03 | INFO | HOSTER ShareonlineBiz[2]: Grabbing link info...
118 | 02.12.2017 13:34:03 | INFO | HOSTER ShareonlineBiz[2]: Link name: TVNowDownloader_Portable_v1.1.0.1.rar
119 | 02.12.2017 13:34:03 | INFO | HOSTER ShareonlineBiz[2]: Link size: 18.20 MiB (19088982 bytes)
120 | 02.12.2017 13:34:03 | INFO | HOSTER ShareonlineBiz[2]: Link status: online
121 | 02.12.2017 13:34:03 | INFO | HOSTER ShareonlineBiz[2]: Processing url: http://www.share-online.biz/dl/U4W59IZO1GT
122 | 02.12.2017 13:34:03 | INFO | HOSTER ShareonlineBiz[2]: Looking for direct download link...
123 | 02.12.2017 13:34:03 | INFO | HOSTER ShareonlineBiz[2]: Direct download link not found
124 | 02.12.2017 13:34:03 | INFO | ACCOUNT ShareonlineBiz: Grabbing account info for user `238*******`...
125 | 02.12.2017 13:34:03 | INFO | HOSTER ShareonlineBiz[2]: Filesize: 18641 KiB \| Traffic left for user `238*******`: 104857600 KiB
126 | 02.12.2017 13:34:03 | INFO | HOSTER ShareonlineBiz[2]: Processing as premium download...
127 | 02.12.2017 13:34:03 | INFO | ACCOUNT ShareonlineBiz: Grabbing account info for user `238*******`...
128 | 02.12.2017 13:34:03 | INFO | HOSTER ShareonlineBiz[2]: Checking download...
129 | 02.12.2017 13:34:03 | INFO | HOSTER ShareonlineBiz[2]: File is OK
130 | 02.12.2017 13:34:03 | INFO | HOSTER ShareonlineBiz[2]: Waiting 5 minutes...

**Comment by GammaC0de** [comment]
If you want me to test this with premium account, please set to a temporary password and then send account user and password to nitzo2001 at yahoo dot com.

**Comment by ironbiff** [comment]
send you a mail

**Comment by GammaC0de** [comment]
OK, This is what I could find so far:

1. It works fine without a docker container.
2. It also works fine in a VM with NAT
3. If you use a docker container it actually downloads a HTML file so this is not a permission issue.

Now the question is why it downloads a HTML file !??

**Comment by GammaC0de** [comment]
OK:
Apparently `writl/pyload:latest` is not so latest and it suffers from #3010 bug.

What you have to do is to either:
- Ask the owner to build an up to date container.
- Build the container yourself (https://github.com/obi12341/docker-pyload)
- Somehow replace with https://github.com/pyload/pyload/raw/stable/module/network/RequestFactory.py

**Comment by ironbiff** [comment]
Woah, thank you for this detailed explanation. 
I thought that every time I reload the docker container the pyload git would updated automatically. Apparently this is not the case. -> I looked into it and now I understand a little bit more of this docker container behaviour!

So first sorry that I disturbed you with this problem and SECOND you will of course get a small donation. You well-deserved it!
